File: SmartPhone++/UI/smartphone_mobile_client/lib/model/cart_item_operation_request.dart
class CartItemOperationRequest {
  final int productId;
  final int quantity;

  CartItemOperationRequest({
    required this.productId,
    required this.quantity,
  });

  Map<String, dynamic> toJson() {
    return {
      'productId': productId,
      'quantity': quantity,
    };
  }
}

File: SmartPhone++/SmartPhone.Model/Requests/CartItemOperationRequest.cs
using System.ComponentModel.DataAnnotations;

namespace SmartPhone.Model.Requests
{
    public class CartItemOperationRequest
    {
        [Required]
        [Range(1, int.MaxValue)]
        public int ProductId { get; set; }
        
        [Required]
        [Range(0, int.MaxValue)] // Allow 0 for removal operations
        public int Quantity { get; set; }
    }
}
